ProgressButton/tests/ProgressButton-specs.js

```diff
@@ -1,77 +1,85 @@
 
 describe('ProgressButton Specs', () => {
 	test('should show text', () => {
-		const expected = 'Progress Button';
-		const progressButton = shallow(
+		const {getByRole} = render(
 			<ProgressButtonBase>
 				Progress Button
 			</ProgressButtonBase>
 		);
-		const actual = progressButton.prop('children');
+
+		const expected = 'Progress Button';
+		const actual = getByRole('button').textContent;
+
 		expect(actual).toBe(expected);
 	});
 
 	test('should have a root className `progressButton`', () => {
-		const expected = 'progressButton';
-		const progressButton = shallow(
+		const {getByRole} = render(
 			<ProgressButtonBase>
 				Progress Button
 			</ProgressButtonBase>
 		);
-		const actual = progressButton.prop('className');
-		expect(actual).toBe(expected);
+
+		const expected = 'progressButton';
+		const actual = getByRole('button').className;
+
+		expect(actual).toContain(expected);
 	});
 
 	test('should show radial progress', () => {
-		const expected = 'radial';
-		const progressButton = shallow(
+		const {getByRole} = render(
 			<ProgressButtonBase showProgress>
 				Progress Button
 			</ProgressButtonBase>
 		);
 
-		const actual = progressButton.find('.progressContainer').childAt(0).prop('orientation');
-		expect(actual).toBe(expected);
+		const expected = 'radial';
+		const actual = getByRole('progressbar').className;
+
+		expect(actual).toContain(expected);
 	});
 
 	test('should show 0.5 progress', () => {
-		const expected = 0.5;
-		const progressButton = shallow(
+		const {getByRole} = render(
 			<ProgressButtonBase showProgress progress={0.5}>
 				Progress Button
 			</ProgressButtonBase>
 		);
 
-		const actual = progressButton.find('.progressContainer').childAt(0).prop('progress');
+		const expected = '0.5';
+		const styleValues = getByRole('progressbar').style._values;
+		const {'--ui-progressbar-proportion-end':actual} = styleValues;
+
 		expect(actual).toBe(expected);
 	});
 
 	test('should show default icon `stop`', () => {
-		const expected = 'stop';
-
-		const progressButton = shallow(
+		const {getByRole} = render(
 			<ProgressButtonBase showProgress progress={0.5}>
 				Progress Button
 			</ProgressButtonBase>
 		);
 
-		const actual = progressButton.find('.icon').prop('children');
-		expect(actual).toBe(expected);
+		const expectedCode = 983004;
+		const actualCode = getByRole('button').textContent.codePointAt();
+
+		expect(actualCode).toBe(expectedCode);
 	});
 
 	test('should support a custom icon', () => {
-		const expected = 'play';
-
-		const progressButton = shallow(
-			<ProgressButtonBase showProgress progress={0.5} icon={expected}>
+		const customIcon = 'star';
+		const {getByRole} = render(
+			<ProgressButtonBase showProgress progress={0.5} icon={customIcon}>
 				Progress Button
 			</ProgressButtonBase>
 		);
 
-		const actual = progressButton.find('.icon').prop('children');
-		expect(actual).toBe(expected);
+		const expectedCode = 983080;
```

Comment:
Add a comment gere: // decimal converted charCode of Unicode 'play' character

Reply:
Done